package MonarchPB;

// The MonarchHeader class in protocol buffer form.
message MonarchHeader {
  // The name of the file this belongs to.  Seems redundant
  // but could be useful in terms of keeping track of files.
  required string filename = 1;

  // The digitizer data is its own little message
  message DigitizerInfo {
    required uint32 digRate = 1;
    required uint32 recSize = 2; 
    required uint32 acqMode = 3;
  }
  required DigitizerInfo digInfo = 2;

  // It's nice to know where data came from.  This flag will
  // specify that the data is either from mantis or it is
  // simulated. 
  enum DataSource {
    Mantis = 0;
    Simulation = 1;
  } 

  // It's also nice to know if you expect this file to be
  // a background measurement or to contain an expected signal.
  enum RunType {
    Background = 0;
    Signal = 1;
  }

  // How long did we digitize for?
  required uint32 acqTime = 3;

  // Now things like creation date (because why not)
  optional string createdate = 4 [default = ""];

  // Optionally flag the source of the data.
  optional DataSource dsource = 5;

  // Optionally flag the content of the data.
  optional RunType rtype = 6;
}